From cd4f24ba2674fdda24bf791f707446ef0dc52e0a Mon Sep 17 00:00:00 2001
From: Huaxiang Sun <hsun@cloudera.com>
Date: Sat, 3 Dec 2016 07:18:36 -0800
Subject: [PATCH 250/308] HBASE-17243 Reuse CompactionPartitionId and avoid
 creating MobFileName in PartitionedMobCompactor to
 avoid unnecessary new objects (addendum)

Signed-off-by: Matteo Bertozzi <matteo.bertozzi@cloudera.com>

Change-Id: Idaa1842dca2108a7c53a437731c6164773232746
Author: Huaxiang Sun
Reason: Improvement
Ref: CDH-43166
---
 .../PartitionedMobFileCompactor.java               |    7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/mob/filecompactions/PartitionedMobFileCompactor.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/mob/filecompactions/PartitionedMobFileCompactor.java
index 51f93e6..c38ccef 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/mob/filecompactions/PartitionedMobFileCompactor.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/mob/filecompactions/PartitionedMobFileCompactor.java
@@ -160,10 +160,11 @@ public class PartitionedMobFileCompactor extends MobFileCompactor {
         id.setDate(MobFileName.getDateFromName(fileName));
         CompactionPartition compactionPartition = filesToCompact.get(id);
         if (compactionPartition == null) {
-          compactionPartition = new CompactionPartition(
-              new CompactionPartitionId(id.getStartKey(), id.getDate()));
+          CompactionPartitionId newId = new CompactionPartitionId(id.getStartKey(), id.getDate());
+          compactionPartition = new CompactionPartition(newId);
+
           compactionPartition.addFile(file);
-          filesToCompact.put(id, compactionPartition);
+          filesToCompact.put(newId, compactionPartition);
         } else {
           compactionPartition.addFile(file);
         }
-- 
1.7.9.5

